service/src/main/java/bio/terra/workspace/app/controller/Alpha1ApiController.java

```diff
@@ -0,0 +1,238 @@
+  }
+
+  // Convert a WsmResource into the API format for enumeration
+  @VisibleForTesting
+  public ApiResourceUnion apiResourceFromWsmResource(WsmResource wsmResource, String gcpProjectId) {
+
+    var union = new ApiResourceUnion();
+    switch (wsmResource.getStewardshipType()) {
+      case REFERENCED:
+        ReferencedResource referencedResource = wsmResource.castToReferencedResource();
+        switch (wsmResource.getResourceType()) {
+          case BIG_QUERY_DATASET:
+            {
+              ReferencedBigQueryDatasetResource resource =
+                  referencedResource.castToBigQueryDatasetResource();
+              union.gcpBqDataset(resource.toApiResource());
+              break;
+            }
+          case BIQ_QUERY_DATA_TABLE:
+            {
+              ReferencedBigQueryDataTableResource resource =
+                  referencedResource.castToBigQueryDataTableResource();
+              union.gcpBqDataTable(resource.toApiResource());
+              break;
+            }
+          case DATA_REPO_SNAPSHOT:
+            {
+              ReferencedDataRepoSnapshotResource resource =
+                  referencedResource.castToDataRepoSnapshotResource();
+              union.gcpDataRepoSnapshot(resource.toApiResource());
+              break;
+            }
+
+          case GCS_BUCKET:
+            {
+              ReferencedGcsBucketResource resource = referencedResource.castToGcsBucketResource();
+              union.gcpGcsBucket(resource.toApiModel());
+              break;
+            }
+
+          case GCS_OBJECT:
+            {
+              ReferencedGcsObjectResource resource = referencedResource.castToGcsObjectResource();
+              union.gcpGcsObject(resource.toApiModel());
+              break;
+            }
+
+          default:
+            throw new InternalLogicException(
+                "Unknown referenced resource type: " + wsmResource.getResourceType());
+        }
+        break; // referenced
+
+      case CONTROLLED:
+        ControlledResource controlledResource = wsmResource.castToControlledResource();
+        switch (wsmResource.getResourceType()) {
+          case AI_NOTEBOOK_INSTANCE:
+            {
+              ControlledAiNotebookInstanceResource resource =
+                  controlledResource.castToAiNotebookInstanceResource();
+              union.gcpAiNotebookInstance(resource.toApiResource(gcpProjectId));
+              break;
+            }
+          case GCS_BUCKET:
+            {
+              ControlledGcsBucketResource resource = controlledResource.castToGcsBucketResource();
+              union.gcpGcsBucket(resource.toApiResource());
+              break;
+            }
+
+          case BIG_QUERY_DATASET:
+            {
+              ControlledBigQueryDatasetResource resource =
+                  controlledResource.castToBigQueryDatasetResource();
+              union.gcpBqDataset(resource.toApiResource(gcpProjectId));
```

Comment:
> Double-checking: will this change break currently active flights with the switch from `JobMapKeys.REQUEST` to `ResourceKeys.RESOURCE` in a couple places? I think that's probably fine given our low usage and that none of the modified flights run for very long, but good to confirm

Yes, I was hoping it would be find. I started leaving both in, but that made it hard to know that I had found all of the places that needed to change.